examples/consumergroup/main.go

```diff
@@ -162,13 +161,16 @@ func toggleConsumptionFlow(client sarama.ConsumerGroup, isPaused *bool) {
 
 // Consumer represents a Sarama consumer group consumer
 type Consumer struct {
-	ready chan bool
+	ready       chan bool
+	readyCloser sync.Once
 }
 
 // Setup is run at the beginning of a new session, before ConsumeClaim
 func (consumer *Consumer) Setup(sarama.ConsumerGroupSession) error {
 	// Mark the consumer as ready
-	close(consumer.ready)
+	consumer.readyCloser.Do(func() {
```

Comment:
Only close the channel once since only the `main()` is waiting on the `ready` channel. No need to close again in cases of rebalance since no one is waiting after first close.